### PR TITLE
prepare for Twig templating

### PR DESCRIPTION
**Pull request by briskt**
### Changed
- Generate the browser.js path in the controller, not the template.
- Pass the announcement text as a template parameter.
- Move PHP code from other_mfas.php template into the prompt-for-mfa.php controller.
- Ensure recaptcha.siteKey is assigned in the controller so a null coalesce isn't needed in the template.

### Fixed
- Fixed the `copyJsLib` Make target
- Fixed an error in the development/hub/metadata/idp-remote.php metadata

### Removed
- Removed extra copy of browser.js

### CODE REVIEW

File: modules/material/themes/material/common-announcement.php [hunk 1]
```diff
@@ -1,15 +1,10 @@
 <?php
-
-use Sil\SspUtils\AnnouncementUtils;
-
-// TODO: remove this
-$announcementHtml = AnnouncementUtils::getAnnouncement();
-
-if (! empty($announcementHtml)) {
+if (! empty($this->data['announcement'])) {
```
Comment (mtompset) [inline modules/material/themes/material/common-announcement.php:2]: should have a blank line after the initial <?php line.

File: modules/material/themes/material/mfa/other_mfas.php [hunk 2]
```diff
@@ -1,23 +1,5 @@
...
-$otherOptions = excludeSelf($mfaOptions, $currentMfaId);
-
-if (! empty($this->data['managerEmail'])) {
-    $otherOptions[] = [
-        'type' => 'manager',
-        'callback' => '/module.php/mfa/send-manager-mfa.php?StateId='.htmlentities($this->data['stateId'])
-    ];
-}
-
+$otherOptions = $this->data['otherOptions'];
```
Comment (mtompset) [inline modules/material/themes/material/mfa/other_mfas.php:2]: add blank line above, please.

File: modules/material/themes/material/mfa/other_mfas.php [hunk 3]
```diff
@@ -29,23 +11,16 @@ function excludeSelf($others, $selfId) {
...
-                <?php
-                $label = empty($option['id']) ? 'help' : $type;
-                ?>
-                <?= $this->t('{material:mfa:use_' . $label . '}') ?>
+                <img
+                  class="mdl-list__item-icon"
+                  src="<?= $option['image'] ?>"
+                  alt="<?= $this->t('{material:mfa:' . $option['type'] . '_icon}') ?>"
+                >
+                <?= $this->t('{material:mfa:use_' . $option['label'] . '}') ?>
```
Comment (mtompset) [inline modules/material/themes/material/mfa/other_mfas.php:23]: Not the same logic as 46. Is there a reason?

File: modules/mfa/www/prompt-for-mfa.php [hunk 4]
```diff
@@ -96,6 +96,25 @@
 
 $globalConfig = Configuration::getInstance();
 
+$otherOptions = array_filter($mfaOptions, function($option) use ($mfaId) {
```
Comment (mtompset) [inline modules/mfa/www/prompt-for-mfa.php:99]: Hadn't see that syntax before! anonymous function with a use.

File: modules/material/themes/material/mfa/other_mfas.php (hunk 3 above)
Comment (briskt) [reply modules/material/themes/material/mfa/other_mfas.php:23]: The logic was moved to modules/mfa/www/prompt-for-mfa.php line 115:

```php
$option['label'] = empty($option['id']) ? 'help' : $option['type'];
```

Brief aside: this is exemplary of the reason I wanted to move the modules all into ssp-base. Prior to that effort, these two files were in two different repositories. It was that much more difficult to synchronize the interface between the modules (e.g. mfa and material) with them in different repos.

File: modules/mfa/www/prompt-for-mfa.php (hunk 4 above)
Comment (briskt) [reply modules/mfa/www/prompt-for-mfa.php:99]: I didn't write it, so I can't take any credit or blame.

**Comment by briskt** [comment]
Thanks for the good review comments and questions, @mtompset.

**Review by mtompset — APPROVED** [review]

**Pull request merged**